pkgs/by-name/ni/niri/package.nix

```
@@ -64,6 +64,22 @@ rustPlatform.buildRustPackage rec {
 
   LIBCLANG_PATH = "${libclang.lib}/lib";
 
+  passthru.providedSessions = ["niri"];
+
+  postInstall = ''
+    mkdir -p $out/share/systemd/user
+    mkdir -p $out/share/wayland-sessions
+    mkdir -p $out/share/xdg-desktop-portal
+
+    cp ${src}/resources/niri-session $out/bin/niri-session
+    cp ${src}/resources/niri.service $out/share/systemd/user/niri.service
+    cp ${src}/resources/niri-shutdown.target $out/share/systemd/user/niri-shutdown.target
+    cp ${src}/resources/niri.desktop $out/share/wayland-sessions/niri.desktop
+    cp ${src}/resources/niri-portals.conf $out/share/xdg-desktop-portal/niri-portals.conf
+  '';
+
+  postFixup = ''sed -i "s#/usr#$out#" $out/share/systemd/user/niri.service'';
```

```suggestion
  postFixup = ''
    sed -i "s#/usr#$out#" $out/share/systemd/user/niri.service
  '';
```

also please use substituteInPlace with --replace-fail, so that we notice when this fails in a future update